File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/geopoint/DistanceMatrixApplicationService.java
package de.helfenkannjeder.helfomat.api.geopoint;

import de.helfenkannjeder.helfomat.core.geopoint.GeoPoint;

public interface DistanceMatrixApplicationService {
    TravelDistanceDto getTravelDistanceFor(TravelModeDto travelMode, GeoPoint origin, GeoPoint destination) throws InvalidTravelModeException;
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/geopoint/DistanceMatrixApplicationService.kt
package de.helfenkannjeder.helfomat.api.geopoint

import de.helfenkannjeder.helfomat.core.geopoint.GeoPoint

interface DistanceMatrixApplicationService {
    fun getTravelDistanceFor(travelMode: TravelModeDto, origin: GeoPoint, destination: GeoPoint): TravelDistanceDto?
}